orcid-persistence/src/main/java/org/orcid/persistence/dao/impl/ProfileDaoImpl.java

```diff
@@ -55,22 +56,51 @@ public void remove(String id) {
         super.remove(id);
     }
 
+    /**
+     * Get a list of the ORCID id's with the given indexing status
+     * @param indexingStatuses
+     *          The list of desired indexing status
+     * @param maxResults
+     *          Max number of results
+     * @return a list of object arrays where the object[0] contains the orcid id and object[1] contains the indexing status                           
+     * */
     @SuppressWarnings("unchecked")
-    public List<String> findOrcidsByIndexingStatus(IndexingStatus indexingStatus, int maxResults) {
+    @Override
+    public List<Pair<String, IndexingStatus>> findOrcidsByIndexingStatus(IndexingStatus indexingStatus, int maxResults) {
         return findOrcidsByIndexingStatus(indexingStatus, maxResults, Collections.EMPTY_LIST);
     }
 
+    /**
+     * Get a list of the ORCID id's with the given indexing status
+     * @param indexingStatuses
+     *          The list of desired indexing status
+     * @param maxResults
+     *          Max number of results
+     * @param orcidsToExclude
+     *          List of ORCID id's to exclude from the results
+     * @return a list of object arrays where the object[0] contains the orcid id and object[1] contains the indexing status                           
+     * */
     @Override
-    public List<String> findOrcidsByIndexingStatus(IndexingStatus indexingStatus, int maxResults, Collection<String> orcidsToExclude) {
+    public List<Pair<String, IndexingStatus>> findOrcidsByIndexingStatus(IndexingStatus indexingStatus, int maxResults, Collection<String> orcidsToExclude) {
         List<IndexingStatus> indexingStatuses = new ArrayList<>(1);
         indexingStatuses.add(indexingStatus);
         return findOrcidsByIndexingStatus(indexingStatuses, maxResults, orcidsToExclude);
     }
 
+    /**
+     * Get a list of the ORCID id's with the given indexing status
+     * @param indexingStatuses
+     *          The list of desired indexing status
+     * @param maxResults
+     *          Max number of results
+     * @param orcidsToExclude
+     *          List of ORCID id's to exclude from the results
+     * @return a list of object arrays where the object[0] contains the orcid id and object[1] contains the indexing status                           
+     * */
     @SuppressWarnings("unchecked")
     @Override
-    public List<String> findOrcidsByIndexingStatus(Collection<IndexingStatus> indexingStatuses, int maxResults, Collection<String> orcidsToExclude) {
-        StringBuilder builder = new StringBuilder("SELECT p.orcid FROM profile p WHERE p.indexing_status IN :indexingStatus");
+    public List<Pair<String, IndexingStatus>> findOrcidsByIndexingStatus(Collection<IndexingStatus> indexingStatuses, int maxResults, Collection<String> orcidsToExclude) {
```

Comment:
@TomDemeranville now we return a list of pairs! List < Pair < String, IndexingStatus > >